client/blb/client_test.go

```diff
@@ -329,7 +316,29 @@ func TestWriteReadTracedExactlyOneTractWithBackups(t *testing.T) {
 	trace.check(t, true, "ts-0000000100000001:0001-1", 1, core.TractLength, 0)
 	trace.check(t, true, "ts-0000000100000001:0001-2", 1, core.TractLength, 0)
 
-	// Send a read with backups. And syncronize backup reads using fake time.
+	// Test each request scenario by overriding backupRequestFunc
+	// Does the first request succeed before backup is sent work?
+	sendPrimary := make(chan bool)
+	sendBackup := make(chan bool)
+	backupRequestFunc = func(
```

Comment:
It might be annoying to have to write a new implementation of this for each test.. I wonder if there's a generic version that can be reused? You can try just writing a few tests and see what commonalities fall out.